### PR TITLE
Fix invalidation log initialization

### PR DESCRIPTION
**Pull request by gayyappan**
The invalidation log initialization did not adjust
the timestamp to internal format. Use ts_get_now_internal
in all places that write entries to the invlaidation log.

### CODE REVIEW

File: tsl/src/continuous_aggs/create.c [hunk 1]
```diff
@@ -514,11 +515,9 @@ mattablecolumninfo_create_materialization_table(MatTableColumnInfo *matcolinfo,
...
 	/* Initialize the invalidation log for the cagg. Initially, everything is
 	 * invalid. */
-	if (OidIsValid(origquery_tblinfo->nowfunc))
-		current_time = ts_time_value_to_internal(OidFunctionCall0(origquery_tblinfo->nowfunc),
-												 get_func_rettype(origquery_tblinfo->nowfunc));
-	else
-		current_time = GetCurrentTransactionStartTimestamp();
+	mat_time_dim = hyperspace_get_open_dimension(ht->space, 0);
+	Assert(mat_time_dim != NULL);
+	current_time = ts_get_now_internal(mat_time_dim);
```
Comment (mkindahl) [inline tsl/src/continuous_aggs/create.c:520]: It looks like the new code does not call the `integer_now` function for the hypertable. Could you elaborate in the commit message why this is actually not the case?

Also, since the `nowfunc` field is filled in from the dimension, do we still need this field?

**Review by mkindahl — APPROVED** [review]
Approving since the code works correctly, but please add information to the commit message so that it is clear that it still uses the `integer_now` function for the table.

File: tsl/src/continuous_aggs/create.c (hunk 1 above)
Comment (gayyappan) [reply tsl/src/continuous_aggs/create.c:520]: ts_get_now_internal handles all the different OID cases for the time dimension. So the change is to use the common api rather than make type specific calls.

**Review by mkindahl — CHANGES_REQUESTED** [review]
You have quite broad failures in the test cases. Can you please have a look at it before I approve it.

**Comment by gayyappan** [comment]
> You have quite broad failures in the test cases. Can you please have a look at it before I approve it.

That's very curious. Worked locally. I'll take a look.

**Review by svenklemm — APPROVED** [review]

File: tsl/src/continuous_aggs/create.c [hunk 2]
```diff
@@ -505,20 +506,23 @@ mattablecolumninfo_create_materialization_table(MatTableColumnInfo *matcolinfo,
...
 	/* Initialize the invalidation log for the cagg. Initially, everything is
-	 * invalid. */
-	if (OidIsValid(origquery_tblinfo->nowfunc))
-		current_time = ts_time_value_to_internal(OidFunctionCall0(origquery_tblinfo->nowfunc),
-												 get_func_rettype(origquery_tblinfo->nowfunc));
-	else
-		current_time = GetCurrentTransactionStartTimestamp();
+	 * invalid.
+	 * Integer time dimensions have now functions. These settings
+	 * are derived from the original ht since they are not explicitly set on
```
Comment (erimatnor) [inline tsl/src/continuous_aggs/create.c:519]:
```suggestion
	 * are derived from the original hypertable since they are not explicitly set on
```

**Review by erimatnor — APPROVED** [review]
I second @mkindahl that there seem to be variables (e.g., nowfunc) that are no longer needed and should be removed.

File: tsl/src/continuous_aggs/create.c (hunk 1 above)
Comment (gayyappan) [reply tsl/src/continuous_aggs/create.c]: removed nowfunc from the struct.

**Review by WireBaron — APPROVED** [review]

File: tsl/test/expected/continuous_aggs_errors.out [hunk 3]
```diff
@@ -473,30 +473,37 @@ select time_bucket(100, timec), min(location), sum(temperature),sum(humidity)
...
+      location    TEXT              NOT NULL,
+      temperature DOUBLE PRECISION  NULL,
+      humidity    DOUBLE PRECISION  NULL,
+      lowp        double precision NULL,
+      highp       double precision null,
+      allnull     double precision null
+    );
+select table_name from create_hypertable( 'conditions', 'timec', chunk_time_interval=> 100);
+ table_name 
+------------
```
Comment (mkindahl) [inline tsl/test/expected/continuous_aggs_errors.out:496]: There is an `\set ON_ERROR_STOP 0` at the beginning here but there are no failing statements after. Is this intentional?

File: tsl/test/expected/continuous_aggs_errors.out [hunk 4]
```diff
@@ -473,30 +473,37 @@ select time_bucket(100, timec), min(location), sum(temperature),sum(humidity)
...
 from conditions
 group by time_bucket(100, timec);
--- Should print a useful error message, but not fail.
-CREATE MATERIALIZED VIEW IF NOT EXISTS mat_with_test
-WITH (timescaledb.continuous) AS
-SELECT time_bucket(100, timec), min(location), sum(temperature),sum(humidity)
-  FROM conditions
-GROUP BY time_bucket(100, timec);
-NOTICE:  continuous aggregate "mat_with_test" already exists, skipping
+ERROR:  invalid integer_now function
```
Comment (mkindahl) [inline tsl/test/expected/continuous_aggs_errors.out:482]: This statement now suddenly starts to fail when it succeeded before. Is this intentional?

File: tsl/test/expected/continuous_aggs_errors.out [hunk 5]
```diff
@@ -473,30 +473,37 @@ select time_bucket(100, timec), min(location), sum(temperature),sum(humidity)
...
 \set ON_ERROR_STOP 0
-ALTER MATERIALIZED VIEW mat_with_test SET(timescaledb.refresh_lag = '1h');
-ERROR:  parameter timescaledb.refresh_lag must be an integer for hypertables with integer time values
-ALTER MATERIALIZED VIEW mat_with_test SET(timescaledb.refresh_lag = '2147483648');
-ERROR:  timescaledb.refresh_lag out of range
-ALTER TABLE conditions ALTER timec type bigint;
-ERROR:  cannot alter type of a column used by a view or rule
-\set ON_ERROR_STOP 1
-DROP MATERIALIZED VIEW mat_with_test;
-ALTER TABLE conditions ALTER timec type bigint;
```
Comment (mkindahl) [inline tsl/test/expected/continuous_aggs_errors.out:499]: The removal of these tests does not seem related to the fix that is implemented. Is this intentional?

**Review by mkindahl — COMMENTED** [review]
The actual fix in the code looks good, but there are some changes in the test cases that I do not understand if they are warranted and/or correct. Could you please check them?

File: tsl/test/expected/continuous_aggs_errors.out (hunk 4 above)
Comment (gayyappan) [reply tsl/test/expected/continuous_aggs_errors.out:482]: yes, current error checking on alter type appears to be inadequate. This si the correct behavior.

File: tsl/test/expected/continuous_aggs_errors.out (hunk 5 above)
Comment (gayyappan) [reply tsl/test/expected/continuous_aggs_errors.out:499]: refresh_lag is going to be removed. (The correct test case would create a table with bigint type and set the integer now func that returns a bigint. This case does not do that.).  So removing the test instead of fixing it.

Comment (gayyappan) [reply tsl/test/expected/continuous_aggs_errors.out:499]: A test case for bigint type follows this one (line 485 onwards).

**Review by mkindahl — APPROVED** [review]
